salt/daemons/masterapi.py

```diff
@@ -587,10 +619,17 @@ def _mine_get(self, load, skip_verify=False):
         minions = _res['minions']
         for minion in minions:
             fdata = self.cache.fetch('minions/{0}'.format(minion), 'mine')
+            if load.get('as_dict'):
+                ret[minion] = {}
             if isinstance(fdata, dict):
-                fdata = fdata.get(load['fun'])
-                if fdata:
-                    ret[minion] = fdata
+                for fun in functions_allowed:
+                    if fdata.has_key(fun):
```

Comment:
this is an anti-pattern - doing a fdata.has_key(fun) then fdata.get(fun). Do you really need it?

Reply:
Just to keep ret[minion][fun] empty if no such mine exists for current minion. Any ideas to make it better?